test/test-BlueprintParser.cc

```diff
@@ -796,11 +796,11 @@ TEST_CASE("Report error when named sub type is referenced in nested members when
     ParseResult<Blueprint> blueprint;
     SectionParserHelper<Blueprint, BlueprintParser>::parse(source, BlueprintSectionType, blueprint, ExportSourcemapOption, Models(), &blueprint);
 
-    REQUIRE(blueprint.report.error.code == MSONError);
-    SourceMapHelper::check(blueprint.report.error.location, 26, 11);
+    REQUIRE(blueprint.report.error.code == Error::OK);
+    REQUIRE(blueprint.report.warnings.empty());
```

Comment:
Here we need to add `REQUIRE`s to check that base type of `A` and `B` are `mson::ImplicitObjectBaseType`.